src/operator/tensor/broadcast_reduce-inl.h

```diff
@@ -278,7 +279,10 @@ MSHADOW_XINLINE void seq_reduce_assign(const index_t idx, const size_t M, const
   Reducer::SetInitValue(val, residual);
   for (size_t k = 0; k < M; ++k) {
     coord = mxnet_op::unravel(k, rshape);
-    Reducer::Reduce(val, AType(OP::Map(big[j + mxnet_op::dot(coord, rstride)])), residual);
+    AType temp = OP::Map(big[j + mxnet_op::dot(coord, rstride)]);
+    if (use_index)
+      memcpy(reinterpret_cast<char*>(&temp), &k, sizeof(size_t));
```

Comment:
Why memcpy?

Reply:
if we use reinterpret_cast<size_t> then there is a compilation error on one ci about type punning

Reply:
Can you paste the error?

Reply:
@leezu 
```
2020-10-13T00:32:03.091Z] /work/mxnet/src/operator/numpy/np_cross.cc:118:61:   required from here
[2020-10-13T00:32:03.091Z] /work/mxnet/src/operator/numpy/./../tensor/./broadcast_reduce-inl.h:284:7: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
[2020-10-13T00:32:03.091Z]        *(reinterpret_cast<size_t*>(&temp)) = k;
[2020-10-13T00:32:03.091Z]        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[2020-10-13T00:32:03.091Z] /work/mxnet/src/operator/numpy/./../tensor/./broadcast_reduce-inl.h: In instantiation of 'void mxnet::op::broadcast::seq_reduce_assign(mxnet::index_t, size_t, bool, const DType*, OType*, const mshadow::Shape<dimension>&, const mshadow::Shape<dimension>&, const mshadow::Shape<dimension>&, const mshadow::Shape<dimension>&) [with Reducer = mxnet::op::mshadow_op::sum; int ndim = 1; AType = double; DType = double; OType = double; OP = mxnet::op::mshadow_op::identity; bool use_index = false; mxnet::index_t = int; size_t = long unsigned int]':
```